solc/CommandLineInterface.cpp

```diff
@@ -590,10 +590,13 @@ bool CommandLineInterface::readInputFilesAndConfigureRemappings()
 					continue;
 				}
 
-				m_sourceCodes[infile.generic_string()] = readFileAsString(infile.string());
+				m_sourceCodes[path] = readFileAsString(infile.string());
 				path = boost::filesystem::canonical(infile).string();
+				sout() << "allowedpath: " << path << "\n";
 			}
 			m_allowedDirectories.push_back(boost::filesystem::path(path).remove_filename());
+			if (!m_basePath.empty())
```

Comment:
Here we also include the base path in the allowed directories.

Reply:
See line 1115

Reply:
For some reason it was not working for me without this.